sw/device/tests/penetrationtests/firmware/ibex_fi.c

```diff
@@ -877,23 +998,343 @@ status_t handle_ibex_fi_char_unconditional_branch(ujson_t *uj) {
   return OK_STATUS();
 }
 
-status_t handle_ibex_fi_char_conditional_branch(ujson_t *uj)
+status_t handle_ibex_fi_char_conditional_branch_beq(ujson_t *uj)
     __attribute__((optnone)) {
   // Clear registered alerts in alert handler.
   sca_registered_alerts_t reg_alerts = sca_get_triggered_alerts();
 
+  uint32_t result1 = 0;
+  uint32_t result2 = 0;
+
   // FI code target.
-  uint32_t branch_if_ = 1;
-  uint32_t branch_else = 0;
   sca_set_trigger_high();
+  asm volatile("addi x5, x0, 0xaf");
+  asm volatile("addi x6, x0, 0xef");
   asm volatile(NOP10);
-  for (int i = 0; i < 10000; i++) {
-    if (branch_if_ > 0) {
-      branch_if_++;
-    } else {
-      branch_else += 10;
-    }
-  }
+  asm volatile(CONDBRANCHBEQ);
```

Comment:
code gen via macros: https://stackoverflow.com/a/28232095 ?

Reply:
The  macro could take the operation `CONDBRANCHBEQ` (or `CONDBRANCHBNE` so it could be used below), etc.